routes/index.js

```diff
@@ -184,53 +188,78 @@ function findTaskDefinition(task, attempt) {
     return ecs.describeTaskDefinition({taskDefinition: task.taskDefinitionArn}).promise().catch(function (err) {
         sleep.sleep(attempt);
         if (attempt <= 5) {
+            //console.log("\n  Reattempting findTaskDefinition for", task.taskDefinitionArn);
             return findTaskDefinition(task, ++attempt);
         } else {
+            //console.log("\n  REJECTing findTaskDefinition for", task.taskDefinitionArn, "after", attempt, "attempts");
             return Promise.reject(err);
         }
     });
 }
 
 function getTasksWithTaskDefinitions(cluster) {
+  console.log("Getting Tasks annotated with Task Definitions for cluster '", cluster, "'...");
   return new Promise(function (resolve, reject) {
     var tasksArray = [];
       listAllTasks(cluster)
           .then(function (allTaskArns) {
               if (allTaskArns.length == 0) {
+                  console.log("\nNo Task ARNs found");
                   resolve([]);
               } else {
-                  var all = allTaskArns.map(function (tasks, index) {
+                  var taskBatches = allTaskArns.map(function (tasks, index) {
                       return index % maxSize===0 ? allTaskArns.slice(index, index + maxSize) : null;
                   }).filter(function(tasks) {
                       return tasks;
                   });
-                  return Promise.all(all.map(function (tasks){
-                      return ecs.describeTasks({cluster: cluster, tasks: tasks}).promise();
-                  }));
+                  return taskBatches;
               }
           })
-          .then(function (describeTasksResponse) {
-                tasksArray = describeTasksResponse.reduce(function(previous, current){
-                    return previous.concat(current.data.tasks);
-                }, []);
-              return Promise.all(tasksArray.map(function (task) {
-                  return findTaskDefinition(task, 1);
+          .then(function (taskBatches) {
+              // Batch the batches :) - describe up to 10 batches of batches of maxSize ARNs at a time
+              // Without batchPromises, we will fire all ecs.describeTasks calls one after the other and could run into API rate limit issues
+              return batchPromises(10, taskBatches, taskBatch => new Promise((resolve, reject) => {
```

Comment:
@karlringrose This sends up to 10 `describeTasks` calls to ECS API with up to 100 tasks each at a time.  The one below sends 100 `describeTaskDefinitions` calls to ECS API at a time.  Chose the numbers based on successful testing.